pyiron_base/jobs/job/extension/server/generic.py

```diff
@@ -441,6 +454,31 @@ def view_queues():
         else:
             return None
 
+    @property
+    def executor(self) -> Union[Executor, None]:
+        return self._executor
+
+    @executor.setter
+    def executor(self, exe: Union[Executor, None]):
+        if isinstance(exe, Executor):
+            self._executor = exe
+            self.run_mode.executor = True
+        elif exe is None:
+            self._executor = None
+            self.run_mode.executor = False
```

Comment:
```suggestion
            self.run_mode.modal = True
```

There is a critical test missing where we turn the executor on and off again -- my original suggestion raises an error! The suggestion here is also not going to cut it, because `executor` is not playing well with the other run modes, but it is at least a step in the right direction.

Problem with my original suggestion:

```python
from concurrent.futures import ProcessPoolExecutor
from pyiron_atomistics import Project

pr = Project('tmp')
pr.remove_jobs(recursive=True, silently=True)

job = pr.create.job.Lammps("lmp")
job.server.executor = ProcessPoolExecutor()
job.server.executor = None
```

gives 

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[7], line 10
      8 job.server.executor = ProcessPoolExecutor()
      9 print(job.server.run_mode)
---> 10 job.server.executor = None
     12 # master = pr.create.job.Murnaghan("murn")
     13 # master.ref_job = job
     14 # master.run()

File ~/work/pyiron/pyiron_base/pyiron_base/jobs/job/extension/server/generic.py:468, in Server.executor(self, exe)
    466 elif exe is None:
    467     self._executor = None
--> 468     self.run_mode.executor = False
    469 else:
    470     raise TypeError(
    471         "The executor has to be derived from the concurrent.futures.Executor class."
    472     )

File ~/work/pyiron/pyiron_base/pyiron_base/jobs/job/extension/server/runmode.py:103, in Runmode.__setattr__(self, name, value)
    101         self.mode = name
    102     else:
--> 103         raise ValueError("A run mode can only be activated using [True].")
    104 else:
    105     super(Runmode, self).__setattr__(name, value)

ValueError: A run mode can only be activated using [True].
```